patches/minecraft/net/minecraft/potion/Potion.java.patch

```diff
@@ -88,6 +88,20 @@
 +        ret.add(new net.minecraft.item.ItemStack(net.minecraft.init.Items.field_151117_aB));
 +        return ret;
 +    }
++    
++    /**
++     * An index that, if different than that of a compared potion effect's potion, will override the default sort order.
++     * If the values are the same, the Vanilla logic alone will be responsible for sort order. Overriding this method is 
++     * useful for potions with changing liquid colors since, without this, the potion effect order would shift in the 
++     * inventory GUI when stacked with others due to Vanilla using liquid color as part of PotionEffect's compareTo logic.
++     * 
++     * @param The PotionEffect instance containing the potion
++     * @return An index to use to sort potion effects if the value is different than that of the other effect's potion
++     */
++    public int getSortIndex(PotionEffect potionEffect)
```

Comment:
Alright, I made the changes. Personally, I would prefer to be able to only override the liquid colour part, but I was told not to alter the Vanilla logic so I can make due with a total override.